linera-sdk/src/lib.rs

```diff
@@ -366,7 +366,32 @@ pub struct QueryContext {
     pub chain_id: ChainId,
 }
 
-/// A message together with routing information.
+/// Resources that an application may spend during the execution of a message.
+#[derive(Clone, Copy, Debug, Default, Deserialize, Eq, PartialEq, Serialize)]
+pub struct Resources {
```

Comment:
I originally placed this in `linera-base` because I figured we'd also need to use it in `linera-execution`. It's not a guest only type, right? We have to send it to the host :thinking: